### PR TITLE
When generating service mappings cast pair to set to avoid duplicate pairs

### PR DESCRIPTION
**Pull request by suddrey-qut**
Fixes issue #257

### CODE REVIEW

File: ros1_bridge/__init__.py [hunk 1]
```diff
@@ -574,7 +574,7 @@ def determine_common_services(
                         ):
                             pairs.append((ros1_srv, ros2_srv))
 
-    for pair in pairs:
+    for pair in set(pairs):
```
Comment (peterpena) [inline ros1_bridge/__init__.py:577]: In `determine_message_pairs`, the pairs are added beforehand and then only new pairs are added to the set; otherwise the for loop continues. A similar construct is done in `determine_common_services`, but all the pairs are added and then, in your fix, the pairs object is passed to `set` to give you a unique set of pairs. 

My question is can we have a solution that follows a similar construct in `determine_message_pairs` where both functions follow a similar implementation? Both functions have similar functionality, and therefore, should have similar implementation. It will help with readibility.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,12 @@
 Changelog for package ros1_bridge
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+0.8.2 (2020-01-17)
+------------------
+* fix building test when ROS 1 diagnostic_msgs is isolated from roscpp (`#236 <https://github.com/ros2/ros1_bridge/issues/236>`_)
+* fix service with custom mapped message field (`#234 <https://github.com/ros2/ros1_bridge/issues/234>`_)
+* Contributors: Dirk Thomas
+
```
Comment (dirk-thomas) [inline CHANGELOG.rst:10]: Please revert these unrelated changes.

File: package.xml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>ros1_bridge</name>
-  <version>0.8.1</version>
+  <version>0.8.2</version>
```
Comment (dirk-thomas) [inline package.xml:5]: Same here.

**Comment by hidmic** [comment]
@suddrey-qut are you still willing to push for this changes?

**Comment by suddrey-qut** [comment]
Hi, yes I will make these changes shortly. Been a bit smashed with work sorry.

**Comment by suddrey-qut** [comment]
Going to recreate this pull request based on master branch.
